packages/grafana-ui/src/types/plugin.ts

```diff
@@ -1,110 +1,10 @@
-import { ComponentClass } from 'react';
-import { ReactPanelPlugin } from './panel';
-import {
-  DataQueryOptions,
-  DataQuery,
-  DataQueryResponse,
-  QueryHint,
-  QueryFixAction,
-  DataQueryError,
-} from './datasource';
-import { SeriesData } from './data';
-
-export interface DataSourceApi<TQuery extends DataQuery = DataQuery> {
-  /**
-   *  min interval range
-   */
-  interval?: string;
-
-  /**
-   * Imports queries from a different datasource
-   */
-  importQueries?(queries: TQuery[], originMeta: PluginMeta): Promise<TQuery[]>;
-
-  /**
-   * Initializes a datasource after instantiation
-   */
-  init?: () => void;
-
-  /**
-   * Main metrics / data query action
-   */
-  query(options: DataQueryOptions<TQuery>): Promise<DataQueryResponse>;
-
-  /**
-   * Test & verify datasource settings & connection details
-   */
-  testDatasource(): Promise<any>;
-
-  /**
-   *  Get hints for query improvements
-   */
-  getQueryHints?(query: TQuery, results: any[], ...rest: any): QueryHint[];
-
-  /**
-   *  Set after constructor is called by Grafana
-   */
-  name?: string;
-  meta?: PluginMeta;
-  pluginExports?: PluginExports;
```

Comment:
This is the main thing I wanted to get rid of. As PluginExports also contained this same interface DataSourceApi and hence needed to be in this file instead of the types/datasource.ts. Now this is replaced with a components prop (see above)

Reply:
Makes sense to me.